api/src/main/java/org/apache/iceberg/metrics/MetricsContext.java

```diff
@@ -66,11 +66,20 @@ default void initialize(Map<String, String> properties) {
      * Reporting count is optional if the counter is reporting externally.
      *
      * @return current count if available
+     * @deprecated Use {@link Counter#value()}
      */
+    @Deprecated
     default Optional<T> count() {
       return Optional.empty();
     }
 
+    /**
+     * Reports the current count.
+     *
+     * @return The current count
+     */
+    T value();
```

Comment:
we could make this a breaking API change and add the justification to RevAPI (as it is here) or (as @danielcweeks 
 suggested) change this to 
```
default T value() {
  throw new UnsupportedOperationException("Count is not supported.");
}
```

Reply:
I'd say let's add the default.